pkg/apis/core/v1/defaults_test.go

```diff
@@ -1919,3 +1921,15 @@ func TestSetDefaultServiceInternalTrafficPolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestSetDefaultAllocationMode(t *testing.T) {
+	class := &v1.ResourceClaim{}
+
+	// field should be defaulted
+	defaultMode := v1.AllocationModeWaitForFirstConsumer
```

Comment:
also it shouldn't change if the field is set

Reply:
Added.